lib/board.rb

```diff
@@ -0,0 +1,65 @@
+class Board
+  attr_reader :grid
+
+  SPACES = 9
+  ROWS = [[0, 1, 2], [3, 4, 5], [6, 7, 8]].freeze
+  COLUMNS = [[0, 3, 6], [1, 4, 7], [2, 5, 8]].freeze
+  DIAGONALS = [[0, 4, 8], [2, 4, 6]].freeze
+
+  def initialize
+    @grid = create
+  end
+
+  def print
+    puts "|#{@grid[0]}|" + "#{@grid[1]}|" + "#{@grid[2]}|"
+    puts "|#{@grid[3]}|" + "#{@grid[4]}|" + "#{@grid[5]}|"
+    puts "|#{@grid[6]}|" + "#{@grid[7]}|" + "#{@grid[8]}|"
+  end
+
+  def space_exist?(space)
+    space >= 0 && space < 9
+  end
+
+  def space_is_free?(space)
+    space_exist?(space) && @grid[space] == '-'
```

Comment:
Coupling: Using an instance variable (`@grid`), even inside an object, couples the code that uses it to the implementation.  If `@grid` becomes a method instead of an instance variable, this line of code will need to change.  Use an `attr_reader` to avoid this.